src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -468,7 +516,9 @@ def add_forward_model(
 def _from_nested_dict(data: Mapping[str, Any]) -> PartialSnapshot:
     partial = PartialSnapshot()
     if "metadata" in data:
-        partial._metadata = data["metadata"]
+        partial._metadata = SnapshotMetadata(
+            **{k: v for k, v in data["metadata"].items() if v}
```

Comment:
I like to avoid using single letter objects. This reads better if k is replaced with key, and v with value.